v1/dictionary/handlers.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+var fluid = require('infusion');
+var AdaptiveContentServices = fluid.registerNamespace('AdaptiveContentServices');
+var wd = require('word-definition');
+
+require('kettle');
+
+/* General grade for dictionary service endpoints
+ * from which all other handler grades will extend
+*/
+fluid.defaults('AdaptiveContentServices.handlers.dictionary', {
+  gradeNames: 'kettle.request.http',
+  invokers: {
+    handleRequest: 'fluid.notImplemented',
+    wikiDefinition: {
```

Comment:
Since the `wikiDefinition` and `checkWikiErr` are specific to the Wiktionary implementation, they should be on that grade definition instead.

What I'd suggest as a pattern is something like the following, which is similar to other "abstract" grades in Kettle:

```
fluid.defaults('adaptiveContentServices.handlers.dictionary', {
  gradeNames: 'kettle.request.http',
  invokers: {
    // Invoker that all kettle requests are expected to have;
    // but this one simply passes to another invoker that
    // will represent the grade-specific handler
    handleRequest: '{that}.dictionaryHandlerImpl',
    // This is what we'll call the grade-specific handler (the "impl" is short for "implementation")
    dictionaryHandlerImpl: 'fluid.notImplemented',
    // We also expect to have code to check for an error from the service
    checkDictionaryErrorImpl: 'fluid.notImplemented'
  }
});

// Specific grade for wiktionary defined here
fluid.defaults('adaptiveContentServices.handlers.dictionary.wiktionary', {
  gradeNames: ['adaptiveContentServices.handlers.dictionary'],
  invokers: {
    dictionaryHandlerImpl: {
      funcName: 'adaptiveContentServices.handlers.dictionary.wiktionary.getDefinition',
      args: ['{arguments}.0', '{that}']
    },
    checkDictionaryErrorImpl: 'adaptiveContentServices.handlers.dictionary.wiktionary.checkDictionaryError'
  }
});

// Implementation of the wiktionary-specific handler 
adaptiveContentServices.handlers.dictionary.wiktionary.getDefinition = function (request, that) {
    // When checking the response for an error, you can do something like the following: 
    var serviceResponse;
    var errorResponse = that.checkDictionaryErrorImpl(serviceResponse);
};

// Implementation of the wiktionary-specific error checking here
adaptiveContentServices.handlers.dictionary.wiktionary.checkDictionaryError = function (serviceResponse) {
    // error-checking of service response code
};
```

Reply:
Okay!
I will try implementing this in the next commit.